main.go

```diff
@@ -119,13 +126,63 @@ func main() {
 	}
 
 	go func() {
-		// Wait for sigterm
 		sigs := make(chan os.Signal, 1)
-		signal.Notify(sigs, syscall.SIGTERM)
+		signal.Notify(sigs,
+			syscall.SIGTERM,
+			syscall.SIGABRT,
+			syscall.SIGHUP,
+			syscall.SIGINT,
+		)
 		<-sigs
 		log.Println("Terminating")
 		cancel()
 	}()
 
 	r.Run(ctx)
 }
+
+func watchUpdates(doRestartOnUpdate, skipUpdateCheckOnStart bool, autoUpdateCheckFrequency time.Duration) {
+	if !skipUpdateCheckOnStart {
+		runUpdate(doRestartOnUpdate)
+	} else {
+		log.Printf("Version update on startup is skipped, next update check is scheduled in %s",
+			autoUpdateCheckFrequency)
+	}
+
+	periodicalUpdateChecker := time.NewTicker(autoUpdateCheckFrequency)
```

Comment:
Even though this function only returns on exiting the program, let's still defer ticker.Stop() call here